### ISSUE TITLE
Fax und Telefon über Fritzbox

### ISSUE THREAD

**Issue by Maz111111** [jlawyerorg/j-lawyer-org#563]
#78
#73
#30
Also es gibt schon eine Menge Einträge. Gerade die ersten zwei - die ich persönlich sehr spannend finde - sind in den Milestone "Backlog" verschoben.
Weil ich jetzt einfach auch die Faxfunktion dazupacke, behandle ich dass hier mal als neu.
Irgendwie scheint das sonst in Vergessenheit zu geraten.
Die Software phoner macht es möglich, über die Fritzbox zu telefonieren. Der Code ist - soweit ich das sehe - frei. In jedem Fall muss es eine Schnittstelle der Fritzbox für solche Szenarien geben.
Nachdem AVM doch einer der Marktführer sein dürfte, halte ich es für sehr sinnvoll (neben #319 und der sipgate Geschichte) zeitnah irgendwas in der Art in jL zu integrieren, womit es möglich ist, die (wohl fast in jeder Kanzlei) vorhandene Fritzbox für Anrufe und Faxe direkt aus der KSW zu nutzen.

Mein aktueller workflow sieht so aus, dass ich über diverse (Maus-) Klicks in der KSW die Telefonnummer auf den Bildschirm hole, diese dann mit Tastatur in den Zwischenspeicher kopiere, dann (meistens wieder mit Maus) zu phoner wechsle, die Nummer mit Maus einfüge und wähle.
Wenn ich dabei dann immer das hübsche Telefonbutton in jL sehe, wünsche ich mir immer entweder obige Möglichkeit(en) oder zumindest eine Funktion die Nummer direkt - ohne "hundert" Klicks - in den Zwischenspeicher zu bekommen.

**Comment by j-dimension** [comment]
> Irgendwie scheint das sonst in Vergessenheit zu geraten.

Es gerät über das Ticketsystem nichts in Vergessenheit. 
Es ist eine Frage der Ressourcen. Eine KSW ist eine umfangreiche Sache, die ich ehrenamtlich und nebenbei mache. Es wäre auch nicht fair, Anwendern eine baldige Lieferung für alle Features zu suggerieren, indem ich es in eine "nahe" Version zuordne :-(

**Comment by PBaumfalk** [comment]
Was ist eine KSW?

**Comment by Maz111111** [comment]
Das ist mir klar. Entschuldige bitte. Du weißt aber aus meinen anderen Kommentaren, dass ich Deine Ressourcenhoheit uneingeschränkt akzeptiere.
Ich dachte, Dein Backlog ist eine Sammlung von Sachen für irgendwann einmal. Und dazu fand ich es vergleichsweise zu wichtig.
Wofür nutzt Du den "Backlog"?
@PBaumfalk
KSW=Kanzleisoftware

**Comment by PBaumfalk** [comment]
Also, wenn die Daten aus der Kanzleisoftware gezogen werden können, dann kann man diese auch per Cron-Job auf die Fritzbox übertragen. Wenn man also die Kontakte ggf. in eine Nextcloud, die mit dem j-lawyer-server installiert wird per Cronjob auf die Fritzbox packt, wäre das Problem gelöst. Die Daten kann man ja schon alle in eine CSV exportieren. Muss dann für die FB nur noch in eine XML-umgewandelt werden.

https://itwelt.org/anleitungen-howto/linux-740817696/617-kontakte-von-nextcloud-owncloud-mit-der-fritz-box-synchronisieren

Und hier hat jemand ein VBS-Script geschrieben:
https://administrator.de/forum/herstellung-telefonbuchs-fritzbox-208836.html

**Comment by j-dimension** [comment]
Ja, Backlog bedeutet "ungeplant". Ich könnte das in eine 2.2 oder sonstwas stecken, aber wir liegen dann so weit in der Vergangenheit dass sich die Pläne sowieso ändern. 

Nichtsdestotrotz: jegliches Feedback nehme ich mir natürlich zu Herzen und wenn ich das Gefühl habe "xyz brauchen scheinbar viele Anwender", dann führt das auch zu Plananpassungen.

**Comment by PBaumfalk** [comment]
Nochmal zum Problemlösungsansatz:

![image](https://user-images.githubusercontent.com/50404378/71724798-e9435280-2e31-11ea-8b6b-dfdc47f89724.png)

Die Fritzbox kann die drei Datenfelder "Telefon, Mobil und Fax" aufnehmen. Name ist eine Verkettung von Name und Vorname oder (bei Gerichten und Behörden) - die Firma.
Die XML-Datei kann man relativ einfach generieren, wenn man aus der CSV eine Datei macht, die Name, Vorname ggf. Firma, Tel, Fax, Mobil in XML umwandelt. Dann kann man das per CronJob in die Fritzbox einfügen und fertig. Oder alternativ über die API in das Telefonbuch der hauseigenen Fritzbox einfügen: https://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/x_contactSCPD.pdf

**Comment by PBaumfalk** [comment]
Obwohl ich das schon wesentlich sexier fände, wenn man die Kontakte in die Nextcloud auslagert und dann in die Fritzbox packt. Dann kann ich auch auf dem iPhone oder dem öFon oder jedem x-beliebigen Telefon auf die Kontakte zugreifen.

**Comment by Maz111111** [comment]
Hmm...geht es jetzt hier noch um das direkte Anrufen aus der KSW heraus oder um die Frage - Nummern in das Telefonbuch der FB zu packen?

**Comment by socram70** [comment]
@PBaumfalk: Hier erscheint es mir ganz sinnvoll einmal darauf hinzuweisen, dass die Fritzbox seit einem der jüngsten Updates endlich auch mit CardDAV umgehen kann. Somit kann man nun auch seine Nextcloud-Kontakte auf die Fritzbox synchronisieren. Das kann im Zusammenspiel damit, dass j-lawyer ja erste Schritte in Richtung Nextcloud-Integration geht, vielleicht eine ganz hilfreiche Info sein.

@Maz111111 : Vermutlich um beides :-) Ich wäre aber auch sehr begeistert, wenn ich aus j-lawyer heraus direkt per Klick einen Anruf tätigen oder faxen könnte. Faxen könnt ihr ja zumindest, wenn ihr Fritzfax auf dem PC installiert - das läuft bei mir auch unter Windows 10 noch anstandslos. Um euch hier viel Arbeit zu sparen, gibt's eine Anleitung mit den Fallstricken, die es zu beachten gilt (insbesondere auch den Kommentar von Sebastian unten): https://it-fvb.de/avm-fritzfax-auf-windows-10-mit-ip-anschluss-installieren-und-faxe-vom-pc-aus-versenden.html/comment-page-10#comments
